retry/retry.go

```diff
@@ -0,0 +1,57 @@
+package retry
```

Comment:
We have a version of this already in `kubergrunt`, so I extracted it out to `go-commons` so we can use it here: https://github.com/gruntwork-io/go-commons/pull/41

Reply:
FYI the `go-commons` retry function is now merged and available! https://github.com/gruntwork-io/go-commons/releases/tag/v0.8.2